File: C++/Labs/Lab5/LowerRegister.cpp
#include "LowerRegister.h"
LowerRegister::LowerRegister(string str)
{
    if (str != "")
    {
        for (size_t i = 0; i < str.length(); i++)
        {
            Str += tolower(str[i]);
        }
    }
}
void LowerRegister::PasteSymbol(char symb) 
{
    Str += "\\";
    Str += tolower(symb);
}
int LowerRegister::GetLength()
{
    int counter = 0;
    if (Str != "")
    {
        for (int i = 0; i < Str.length(); i++)
        {
            if (Str[i] != '\\') counter++;
        }
    }

    return counter;
}

File: CSharp/CSharp/Lab5/LowerRegister.cs
using System;

namespace Lab5
{
    public class LowerRegister : String
    {
        public LowerRegister(string str):base(str)
        {
            Str = Str?.ToLower();
        }

        public override void PasteSymbol()
        {
            if (Str != null)
            {
                for (int i = 1; i < Str.Length; i += 2)
                {
                    Str = Str.Insert(i, "\\");
                }
            }
            else
            {
                Str += "\\";
            }
        }
        public override int GetLength()
        {
            int counter = 0;
            if (Str != null)
            {
                foreach (var symb in Str)
                {
                    if (symb != '\\') counter++;
                }
            }
            return counter;
        }
    }
}